### PR TITLE
Ethereum Authenticator/Signer

### PR DESCRIPTION
**Pull request by BurntVal**
SIGNING DEMO:

https://github.com/burnt-labs/xion.js/assets/117188229/b4408c8b-289c-46f0-8a6d-d5371de49c90

### CODE REVIEW

**Comment by BurntVal** [comment]
Need to add detection for both Keplr and Metamask so that we can properly handle an account change via extension

**Comment by BurntVal** [comment]
Advanced Options toggle and account detection demo:


https://github.com/burnt-labs/xion.js/assets/117188229/d8babba0-c0ac-4ea8-b133-8e658c063ad7

**Review by ash-burnt — APPROVED** [review]

File: apps/abstraxion-dashboard/hooks/useAbstraxionSigningClient.ts [hunk 1]
```diff
@@ -35,59 +36,81 @@ export const useAbstraxionSigningClient = () => {
     undefined,
   );
 
-  useEffect(() => {
-    async function getSigner() {
-      let signer: AbstractAccountJWTSigner | AADirectSigner | undefined =
-        undefined;
+  async function ethSigningFn(msg: any) {
+    const accounts = await window.ethereum.request({
```
Comment (justinbarry) [inline apps/abstraxion-dashboard/hooks/useAbstraxionSigningClient.ts:40]: Will this fail if `window.ethereum` is undefined?

**Review by justinbarry — APPROVED** [review]
LGTM

**Pull request merged**